blog/bloom-and-cuckoo.md

```diff
@@ -0,0 +1,158 @@
+Fig. 4 shows the results. The insertion performance is constant across all levels of occupancy
+for Bloom filters. For Cuckoo filters, the performance decreases as the filter becomes fuller because more relocations are required. Note that in Fig. 4, the performance for Bloom
+filters is actually not constant at first. It quickly increases and then remains constant. This can be explained by CPU caching. 
+
+<p align="center">
+<img src="./assets/bloom-cuckoo/insert.png">
+</p>
+<p align="center">
+<i>Figure 4: Insertion performance for different occupancy levels. The shaded region indicates the standard deviation.</i>
+</p>
+
+## Capacity and Scaling
+We already mentioned the capacity of a filter several times now. An interesting case to consider is what happens when a filter's capacity is exceeded.
+Bloom filters and Cuckoo filters behave differently in this scenario. For Bloom filters, the `insertion` operation always succeeds. However, the false positive rate
+will rapidly increase as we exceed the filter's capacity. While Bloom filters fail silently, Cuckoo filters are more explicit. Most implementations have a maximum number of
+relocations that will be performed for an insertion. If more relocations are required, the `insertion` operation will return an error.  
+For both filters, we can avoid this problem by simply initializing the filter with a sufficiently large capacity. However, this will increase the memory footprint of the filter.
+Furthermore, it is difficult to predict how many elements a node on Fleek Network will cache. It is also likely that the number of cached elements will greatly vary for different nodes.  
+Fortunately, there is a variant of Bloom filters called _Scalable Bloom Filters_ [4] that can adapt dynamically to the number of elements stored, while guaranteeing a maximum false positive rate.
+The proposed technique is also applicable to Cuckoo filters.
+
+## Other Filters
+While we only looked at Bloom filters and Cuckoo filters, there are other AMQ filters that we want to briefly mention here:
+* Quotient filters [5, 6]: Compact hash tables that support insertion, lookup, and deletion. Less space-efficient than Bloom filters and Cuckoo filters.  
+* XOR filters [7]: More space-efficient than Bloom filters and Cuckoo filters. However, they are static meaning the filter has to be rebuilt if additional elements are added.  
+
+## Conclusion
+We examined whether Counting Bloom filters or Cuckoo filters are more suitable for summarizing caches on Fleek Network. Cuckoo filters are more space-efficient, especially
+for lower false positive rates. Bloom filters have a slightly better insertion and lookup performance, at least for the implementations that we tested.
+Both filters can be adapted to dynamically grow and shrink in size as needed. Since the difference in insertion and lookup performance is negligible while Cuckoo filters
```

Comment:
```suggestion
Both filters can be adapted to grow and shrink in size dynamically. Since the difference in insertion and lookup performance is negligible while Cuckoo filters
```